src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -1459,21 +1448,22 @@ private string GetStackTrace()
             }
             workflowComplete = completionCommandIndex >= 0;
 
-            // This only applies for successful activations that have a completion not at the end,
-            // don't have completion reordering disabled, and that are either not replaying or have
-            // the flag set.
+            // In a previous version of .NET, if this was a successful activation completion with
+            // a completion command not at the end, we'd reorder it to move at the end. However,
+            // this logic has now moved to core and become more robust. Therefore, we only apply
+            // this logic if we're replaying and flag is present so that workflows/histories that
+            // were created after this .NET flag but before the core flag still work.
             if (completion.Successful == null ||
                 completionCommandIndex == -1 ||
                 completionCommandIndex == completion.Successful.Commands.Count - 1 ||
-                disableCompletionCommandReordering ||
-                (IsReplaying && !act.AvailableInternalFlags.Contains(
-                    (uint)WorkflowLogicFlag.ReorderWorkflowCompletion)))
+                !IsReplaying ||
+                !act.AvailableInternalFlags.Contains((uint)WorkflowLogicFlag.ReorderWorkflowCompletion))
             {
                 return;
             }
 
-            // Now we know the completion is in the wrong spot and we're on a newer SDK, so set the
-            // SDK flag and move it
+            // Now we know that we're replaying w/ the flag set and the completion in the wrong
+            // spot, so set the SDK flag and move it
             completion.Successful.UsedInternalFlags.Add((uint)WorkflowLogicFlag.ReorderWorkflowCompletion);
             var compCmd = completion.Successful.Commands[completionCommandIndex];
             completion.Successful.Commands.RemoveAt(completionCommandIndex);
```

Comment:
Now consider a workflow born after this PR is released. Again the raw commands collected in lang are `N C1 N C2 N`. lang does not do anything to them, since lang's flag is not set, and core receives them unchanged. Core then does its new behavior, and changes them to `N N N C1`, and sets core's flag, so that this will also be done on replay.

Reply:
Yes, everything in your statements are correct